# Video Game Sales

Importe todos los documentos que se requieren para el proyecto

In [2]:
import seaborn as s
import numpy as n
import pandas as p

Se uso pandas (p) para abrir datos seleccionados 

In [3]:
%ls Video_Games_Sales_as_at_22_Dec_2016.csv

Video_Games_Sales_as_at_22_Dec_2016.csv


In [5]:
df=p.read_csv("Video_Games_Sales_as_at_22_Dec_2016.csv")

In [6]:
df.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN

Se quitaron variables que no infuian ya que estas se demostraban cuando el juego habia tenido opinion del publico.

In [7]:
df=p.read_csv("Video_Games_Sales_as_at_22_Dec_2016.csv")
df.drop("User_Count",
        axis=1,inplace=True)
df.drop("Critic_Count",
        axis=1,inplace=True)
df.drop("User_Score",
        axis=1,inplace=True)
df.drop("Critic_Score",
        axis=1,inplace=True)
df
df.head()



Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales Developer Rating  
0     41.36     28.96      3.77         8.45         82.53  Nintendo      E  
1     29.08      3.58      6.81         0.77         40.24       NaN    NaN  
2     15.68     12.76      3.79         3.29         35.52  Nintendo      E  
3     15.61     10.93      3.28         2.95         32.77  Nintendo      E  
4     11.27      8.89     10.22         1.00         31.37       NaN    NaN

La variable de interes son las Global_Sales y son impactadas por las demas variables dando valor agregado dependiendo de su Genere, Platform, Developer, Rating entre otras ya que este da diferentes alcances a audiencias lo cual afecta la cantidad de personas que compran el juego o bien tienen interes sobre el mismo.

Con df.info() vemos que tipo de informacion tenemos (ya sea numerica o bien alfabetica)

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Developer        10096 non-null  object 
 11  Rating           9950 non-null   object 
dtypes: float64(6), object(6)
memory usage: 1.5+ MB


Se aprecian 12 variables y 16719 observaciones que varian entre object y float con 6 de cada una. Sera importante ver que tan drastico sera tener valores numericos en los float para asi tener resultados mas precisos si es que es el caso.

In [20]:
df.describe()

Year_of_Release      NA_Sales      EU_Sales      JP_Sales  \
count     16450.000000  16719.000000  16719.000000  16719.000000   
mean       2006.487356      0.263330      0.145025      0.077602   
std           5.878995      0.813514      0.503283      0.308818   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2020.000000     41.360000     28.960000     10.220000   

        Other_Sales  Global_Sales  
count  16719.000000  16719.000000  
mean       0.047332      0.533543  
std        0.186710      1.547935  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.030000      0.470000  
max       10.570000     82.530000

Con df.describe() se arroja una serie de informacion en donde tomamos en cuenta la media y sus valores mas signifiantes como insignificantes. Me interesa Global_Sales ya que las demas columnas son pequenas partes (excepto year of release) de esta misma seccion por lo cual tomare en cuenta esa seccion solamente en este caso. Me di cuenta que otras columnas no son numericas por lo cual debo de hacer un cambio para encontrar su relacion con la influencia en la popularidad de los videojuegos. 

In [ ]:
s.scatterplot(x="Year_of_Release", y="Global_Sales", data=df)

In [ ]:
df['Global_Sales'].plot(kind='hist',figsize=(20,10))

Viendo mis graficas e histograma puedo concluir que es necesario hacer estos cambios en la base de datos, es decir cambiar los floats a datos numericos para sacar relaciones mas precisas ya que se aprecia un histogramaasimetrica en la derecha. Relacione las ventas globales con el ano en el cual estos juegos fueron sacados para determinar su popularidad en el momento y ver si la comunidad respondia de manera diferente a lo largo de los anos. Me interesa ver la relacion de las ventas con la consola o bien con los desarrolladores que hacen los juegos para asi tener un entendimiento mas amplio de los factores que afectan y cual tiene una aceptacion mas alta por la sociedad. 